### PR TITLE
Chaosvolt's weapon weight fixes, and a weapon addition

### PR DESCRIPTION
**Pull request by chaosvolt**
Couple weapons I added without remembering to look up their weight, one
of which was pointed out by AK-74 by the forums.

Weight fixes:
1. BAR was accidentally left at the same weight as the M1903. Doh.
2. AN-94 apparently should be a tad heavier than AK-74.

Additions:
1. Makeshift combination gun, which also adds an unused niche in that we
lack a pipe .30-06 last I checked. Was gonna add a non-makeshift version
before seeing that it's already been done. Neat.

### CODE REVIEW

**Comment by chaosvolt** [comment]
Also, to add a proposal, quoting myself on the forums.

>I'd like to make a proposal. When I first added the Sten, the idea of making it craftable was shot down. First due to the idea that the tools needed are beyond a survivor's means, then balance reasons once a way to craft it with hand tools was pointed out.

If we could make the process of making a Sten a bit more complex, though...my proposal is to allow making an even LOWER quality, semi-auto replica as the first step. This version would have a mod slot for the full-auto conversion part, so getting a full-auto Sten requires at least two steps, and the end product could still be made statistically inferior to the real deal.

I still like the idea of some form of full-auto weapon being craftable, but it needs to be balanced. Any other possible ways this could be made feasible?

**Comment by kevingranade** [comment]
The main thing that would convince me would be to populate a gunsmith shop
with suitable tools for the various operations needed.  I'd still be
pushing for resulting actions being relatively primitive in comparison to
industrially produced guns, but it would expand the reach of the survivor a
bit.

**Comment by chaosvolt** [comment]
Possibly. I would like the idea of a cruder equivalent that can be made with the tools currently available, but it might have to broken up into stages. And should absolutely be time-consuming.

I do think that making a semi-auto version of the Sten and it having a slot for the full-auto conversion might be the easiest solution. I forget, are full-auto conversions even craftable?

**Comment by karlnp** [comment]
And ear plugs.

**Comment by chaosvolt** [comment]
Yes. Though really, the main thing is that nearly every indoor gun range I've seen has an airlock of sorts. Do not open one door unless the second is closed, etc.

Then again...wait, does that work in game terms? Would you still be deafened by gunshots on the other side of, say, reinforced glass walls?

And back on topic, I'm still pondering the "make a semi-auto Sten copy, then drop a full-auto conversion into it" idea.

**Comment by kevingranade** [comment]
Walls don't block sound, that would be super expensive to calculate, you
have to either do pathing or a flood-fill, they're both really expensive
since you need to determine effective distance from every sound source to
every listener.

**Comment by chaosvolt** [comment]
Ah, dang. Oh well, it would still be a nice flavor detail, but beyond the scope of this PR. XP

**Comment by chaosvolt** [comment]
What is within the scope of this PR though, someone suggested the Kel-Tec KSG. We could logically mimic it using built_in_mods to make it...a shotgun, with a shotgun on it.

File: data/json/recipes/recipe_weapon.json [hunk 1]
```diff
@@ -1368,6 +1368,31 @@
   ]
 },{
   "type" : "recipe",
+  "result": "pipe_combination_gun",
```
Comment (kevingranade) [inline data/json/recipes/recipe_weapon.json]: Don't we have enough materials to make this recipe a little more interesting?
The pipes and wood cover the barrels and stock, the scrap covers a trigger assembly, but you need a good spring for the firing pin, which speaking of should probably be a few nails just for ambiance, then you need the breech, that needs to be a block of metal, not just some random scrap, I guess scrap covers misc screws as well, but it'd be better if we had inventories of screws that get used up by recipes like these.  Additionally you need something to bind the whole mess together, rope, duct tape, something.

**Comment by chaosvolt** [comment]
Hmm, maybe. What all do compably complex homemade guns used?

A spring and a nail would be good additions to it though, yeah. Unless about the breech though. You can rather obviously tell it was just an edited version of the double-barrel pipe gun recipe. @_@

**Comment by kevingranade** [comment]
Yes, I know, if you're going to be bulking up the selection, add to the interesting-ness of the recipe too, the pipe guns are boring. :)

**Comment by chaosvolt** [comment]
Ah, sure. Spring and nail at the minimum, and either...hmm. Chunk of steel? Sheet metal? More scrap? Not sure how to suggest for such breech.

**Comment by chaosvolt** [comment]
In other news, while working on statting and testing a KSG prior to adding it, I forget that the weapon mod's to_hit refers to its accuracy when wielded, not a modifier to the gun's to-hit. Doh.

**Comment by chaosvolt** [comment]
Hmm. At least two levels higher crafting requirements than the highest-level makeshift firearm, booklearned only. Need fine sawing, fine screwdriving, fine bolt turning, drilling. Should require a massive time investment to to lack of machining tools. And finally, in addition to the usual makeshift gun parts, you'll need an autofire mechanism. The result should have inferior durability, dispersion, noise, possibly even recoil and damage to a proper Sten.

I would assume if the end result still can't be considered balanced after hitting it with that many requirements and penalties, it just isn't going to be craftable without a metal shop, regardless of there existing methods to make a Sten with tools already in-game.

One final note, last I checked the autofire mechanism is NOT craftable. This places an additional restriction on a player's access to homemade Stens if this part is needed to make it.

**Comment by kevingranade** [comment]
There must have been miscommunication somewhere, semi-auto actions are also
considered beyond the capability of survivors to craft with hand tools, so
a semi-auto Sten with an upgrade path to full-auto is also not ok in
mainline.
The best you're going to do is a multi-barrel gun that chambers one round
per barrel.

**Comment by chaosvolt** [comment]
Hmm. Then I'm assuming the other survival rifles, like the .223 carbine, are supposed to be be manual firearms then?

While I admit that "if X is in,Y should be in" is a stupid argument, saying "X is not in, therefore Y should not be in" makes just as little sense if X is, in fact, in. :V

**Comment by chaosvolt** [comment]
My apologies for this, but the entire cause of my annoyance over this is most likely due a misunderstanding due to an in-game error.

semi-auto lever-action or something

Whatever combination of properties causes the game to declare this thing to be a semi-automatic firearm should presumably be tweaked, as that what the root cause of my misunderstanding here.

**Comment by kevingranade** [comment]
I'd forgotten about that one, I guess that's arguably better than a multi-barrel repeater, though the multi-barrel should fire faster.
The comment from the source is informative:
``` "//": "Homemade tube magazine, based roughly off of early Winchester repeating rifles and 223 round size",```
I'll tack "lever-action" onto the description, does that make it clear?

**Comment by chaosvolt** [comment]
Maybe. Though as I remarked last time the survivor's carbine came up in discussion, that comment just causes even more headaches than it alleviates. .223 Remington. Spitzer bullets. Lever-action. Tube magazine.

Need I say more? =w=

**Comment by kevingranade** [comment]
Yes actually, you do.  I'm guessing you mean something about the way the parts fit together (or don't fit together), but I don't know what.
However, is that at all relevant to the discussion at hand?  There isn't a survivor-craftable semi-auto gun, that's the line we aren't crossing.  This has been the case for well over a year now, and the only exceptions since long before that were legacy items from whales' days that it was inconvenient to remove.

**Comment by chaosvolt** [comment]
Mostly a nitpick about the fact that modern pointed bullets, lined up end-to-end in a tube magazine, and centerfire primers do not mix. And as I've said before, while it is silly of me to complain about a failure to understand how guns work that doesn't affect gameplay and is only visible when eyeballing the json files...

Still, no semi-autos. Understood. Now that I can tell that something seems to be labeling most guns in the game as semi-auto when they aren't, it's fine now. I just kept thinking that you were saying "you can't craft semi-autos" despite their existing craftable semi-autos, but that issue seems to be a simple quirk of gameplay, not the mistake I thought it was. Sorry.

**Comment by kevingranade** [comment]
An additional confusing element is that the lever-actions don't ACT like lever-actions, ideally they should have a flag saying so, and they should reset your aim level in between shots since you have to work the action in between and then reacquire your target.

**Comment by kevingranade** [comment]
Oh THAT, hrm, I didn't even notice the tag there until just now, I don't know what's doing that.  Probably everything that has a magazine gets that.

**Comment by chaosvolt** [comment]
That would be a nice feature, yes.

I should note that logically, a makeshift box magazine should be reasonably craftable compared to tube magazines. Spring, magazine follower, ensure the magazine and follower fit together, then the magazine lips are simple enough once you understand how they work. Since comments seem to primarily be used to explain and clarify why an item's gameplay attributes are the way they are, so a comment like that is both unnecessary and fails to serve its purpose. Still, it is a stupid nitpick on my part, and I admit it. ._.

And yeah, could be that. I've still yet to puzzle out what combination of features makes the game decide a pistol is a revolver. Pistol skill, single reloads, and eject-on-reload I'm assuming?

**Comment by kevingranade** [comment]
Look at the code, pretty sure all that is in item::info()

**Comment by chaosvolt** [comment]
Ah, I see. o.O

>        if (mod->burst_size() == 0) {
            if (skill == Skill::skill("pistol") && has_flag("RELOAD_ONE")) {
                dump->push_back(iteminfo("GUN", _("Revolver.")));

Ahah. Wait, it hinges on having single reloads but not eject-on-reload? Dang, no Webley for me. ;w;

**Comment by chaosvolt** [comment]
And thank you for the merge.

**Pull request merged**